vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -1,207 +1,259 @@
+     * diging.vspace.core.model.ISpace,edu.asu.diging.vspace.core.model.impl.VSImage)
+     */
+    @Override
+    public CreationReturnValue storeSpace(ISpace space, IVSImage image) {
+        List<SpaceDisplay> displays = null;
+        if (space.getId() != null) {
+            displays = spaceDisplayRepo.getBySpace(space);
+        }
+        ISpaceDisplay spaceDisplay;
+        if (displays == null || displays.isEmpty()) {
+            spaceDisplay = spaceDisplayFactory.createSpaceDisplay();
+        } else {
+            spaceDisplay = displays.get(0);
+        }
+
+        if (image != null) {
+            space.setImage(image);
+            spaceDisplay.setHeight(image.getHeight());
+            spaceDisplay.setWidth(image.getWidth());
+        }
+        
+        CreationReturnValue returnValue = new CreationReturnValue();
+        returnValue.setErrorMsgs(new ArrayList<>());
+
+        space = spaceRepo.save((Space) space);
+        spaceDisplay.setSpace(space);
+        spaceDisplayRepo.save((SpaceDisplay) spaceDisplay);
+        returnValue.setElement(space);
+        return returnValue;
+    }
+
+    @Override
+    public ISpace getSpace(String id) {
+        Optional<Space> space = spaceRepo.findById(id);
+        if (space.isPresent()) {
+            return space.get();
+        }
+        return null;
+    }
+
+    @Override
+    public ISpace getFullyLoadedSpace(String id) {
+        ISpace space = getSpace(id);
+        if (space == null) {
+            return null;
+        }
+        // load lazy collections
+        space.getSpaceLinks().size();
+        space.getModuleLinks().size();
+        space.getExternalLinks().size();
+        return space;
+    }
+
+    @Override
+    public List<ISpace> getAllSpaces() {
+        List<ISpace> spaces = new ArrayList<>();
+        spaceRepo.findAll().forEach(s -> spaces.add(s));
+        return spaces;
+    }
+    
+    
+    @Override
+    public List<ISpace> getSpacesWithStatus(SpaceStatus status) {
+        List<ISpace> spaces = new ArrayList<>();
+        spaceRepo.findAllBySpaceStatus(status).forEach(s -> spaces.add(s));
+        return spaces;
+    }
+
+	/**
+	 * Method to delete space based on id
+	 * 
+	 * @param id if id is null throws exception, else delete corresponding space
+	 * @throws SpaceDoesNotExistException
+	 */
+    @Override
+    public void deleteSpaceById(String id) throws SpaceDoesNotExistException {
+        try {
+            // Order of deletion: Tables: SPACELINKDISPLAY (link_id), SPACELINK
+            // (source_space_id), SPACEDISPLAY (space_id), SPACE(id)
+
+            List<SpaceLink> spaceLinks = spaceLinkRepo.getLinkedSpaces(id);
+            List<SpaceLink> fromSpaceLinks = spaceLinkRepo.getLinkedFromSpaces(id);
+            Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition();
+            // When space has other links attached to it
+            if (spaceLinks.size() > 0) {

```

Comment:
I don't think you need this line, especially since in line 228 you are removing the space from incoming links (which wouldn't be called if there are no outgoing links).